src/main/java/org/jnode/fs/xfs/inode/INode.java

```diff
@@ -393,27 +397,70 @@ public String getSymLinkText() {
      *
      * @return list of inode attributes.
      */
-    public List<XfsAttribute> getAttributes() throws IOException {
+    public List<FSAttribute> getAttributes() throws IOException {
```

Comment:
Can you break up this function into smaller functions?

Reply:
Broke it into 2 classes